include/ppx/scene/scene_scene.h

```diff
@@ -24,6 +24,9 @@
 namespace ppx {
 namespace scene {
 
+template <typename ResObjT>
+using ResourceArrayIndexMap = std::pair<std::vector<const ResObjT*>, std::unordered_map<const ResObjT*, uint32_t>>;
```

Comment:
Could this be a struct with meaningful names and comments? 
Or at least a comment on top of this.

Reply:
Simplified and added comment.